src/Enemy/EnemyCap.cpp

```diff
@@ -0,0 +1,186 @@
+}
+
+void EnemyCap::calcAnim() {
+    if (al::isNerve(this, &NrvEnemyCap.Wait)) {
+        updatePose();
+    }
+    al::LiveActor::calcAnim();
+}
+
+void EnemyCap::exeWait() {
+    if (!syncHostVisible())
+        return;
+    al::setModelAlphaMask(this, al::getModelAlphaMask(mCap));
+}
+
+bool EnemyCap::syncHostVisible() {
+    return al::updateSyncHostVisible(&mCapVisible, this, mCap, field_142);
+}
+
+void EnemyCap::exeBlowDown() {
+    if (al::isFirstStep(this)) {
+        al::onCollide(this);
+        al::offSyncClippingSubActor(mCap, this);
+        al::offSyncAlphaMaskSubActor(mCap, this);
+    }
+    al::rotateQuatXDirDegree(this, -15.0f);
+    if (!al::updateNerveState(this))
+        return;
+    al::offCollide(this);
+    al::startHitReaction(this, "消滅");
+    al::onSyncClippingSubActor(mCap, this);
+    al::onSyncAlphaMaskSubActor(mCap, this);
+    kill();
+}
+
+void EnemyCap::startBlowDown(const al::HitSensor* source) {
+    if (isBlowDown())
+        return;
+    mStateBlowDown->start(source);
+    al::setNerve(this, &NrvEnemyCap.BlowDown);
+}
+
+void EnemyCap::startBlowDown() {
+    if (isBlowDown())
+        return;
+    mStateBlowDown->start(mCap);
+    al::setNerve(this, &NrvEnemyCap.BlowDown);
+}
+
+bool EnemyCap::isBlowDown() const {
+    return al::isNerve(this, &NrvEnemyCap.BlowDown);
+}
+
+void EnemyCap::setBlowDownParam(const al::EnemyStateBlowDownParam* param) {
+    mStateBlowDown->setParam(param);
+}
+
+namespace rs {
+EnemyCap* tryCreateEnemyCap(al::LiveActor* actor, const al::ActorInitInfo& info) {
+    const char* str = 0;
+    al::tryGetStringArg(&str, info, "CapName");
+    return rs::tryCreateEnemyCapSuffix(actor, info, str, str);
+}
+
+EnemyCap* tryCreateEnemyCap(al::LiveActor* actor, const al::ActorInitInfo& info, const char* archiveName) {
+    return rs::tryCreateEnemyCapSuffix(actor, info, archiveName, archiveName);
+}
+
+EnemyCap* tryCreateEnemyCapSuffix(al::LiveActor* actor, const al::ActorInitInfo& info, const char* archiveName, const char* suffix) {
+    if (!archiveName)
+        return 0;
+    auto cap = new EnemyCap("帽子");
+    if (!al::isExistSubActorKeeper(actor))
+        al::initSubActorKeeperNoFile(actor, info, 1);
+    cap->initPartsFixFile(actor, info, archiveName, suffix);
+    al::registerSubActor(actor, cap);
+    al::onSyncClippingSubActor(actor, cap);
+    al::onSyncAlphaMaskSubActor(actor, cap);
+    return cap;
+}
+
+bool tryStartEnemyCapBlowDown(EnemyCap* cap, const al::HitSensor* sensor) {
+    if (!isOnEnemyCap(cap))
+        return false;
+    if (cap->isBlowDown())
+        return true;
+    cap->getStateBlowDown()->start(sensor);
+    al::setNerve(cap, &NrvEnemyCap.BlowDown);
+    return true;
+}
+
+bool tryStartEnemyCapBlowDown(EnemyCap* cap) {
+    if (!isOnEnemyCap(cap))
+        return false;
+    if (cap->isBlowDown())
+        return true;
+    cap->getStateBlowDown()->start(cap->getCap());
+    al::setNerve(cap, &NrvEnemyCap.BlowDown);
+    return true;
```

Comment:
```suggestion
    tryStartEnemyCapBlowDown(cap, cap->getCap());
```

Reply:
The cap variable is not a HitSensor so it cannot be passed in.

Reply:
Alright, ignore.